### PR TITLE
Update split library.

### PR DESCRIPTION
**Pull request by jacob314**
Fixes https://github.com/flutter/devtools/issues/864

Fix the issue by providing support for splitting based on specifying width/height as a fixed percentage rather than depending on flex layout which doesn't appear to have the right primitives to support this for cases where the items being flexed need to be shrunk to fit.

Also fixed bug where were passing in minSize but missing the parameter to make them actually be used.

### CODE REVIEW

File: third_party/packages/split/lib/split.dart [hunk 1]
```diff
@@ -40,11 +42,22 @@ class _SplitOptions {
   external List<num> get sizes;
 
   external List<num> get minSize;
+
+  /// minSize is ignored unless this property is set.
```
Comment (natebosch) [inline third_party/packages/split/lib/split.dart]: How about "Whether to use the [minSize] property."

https://dart.dev/guides/language/effective-dart/documentation#prefer-starting-variable-getter-or-setter-comments-with-noun-phrases

File: third_party/packages/split/lib/split.dart [hunk 2]
```diff
@@ -88,31 +107,80 @@ Splitter flexSplit(
       gutterSize: gutterSize,
       sizes: sizes,
       minSize: minSize,
+      expandToMin: minSize?.isNotEmpty == true,
```
Comment (natebosch) [inline third_party/packages/split/lib/split.dart]: [nit] https://dart.dev/guides/language/effective-dart/usage#do-use--to-convert-null-to-a-boolean-value

`expandToMin: minSize?.isNotEmpty ?? false,`

File: third_party/packages/split/lib/split.dart (hunk 1 above)
Comment (kenzieschmoll) [inline third_party/packages/split/lib/split.dart]: nit: [minSize]

File: third_party/packages/split/lib/split.dart [hunk 3]
```diff
@@ -57,8 +70,14 @@ class Splitter {
   external void destroy([bool preserveStyles, bool preserveGutters]);
 }
 
-/// Splitter that splits multiple elements that must be styled with flexbox
-/// layout.
+/// Splitter that splits multiple elements using flex layout.
+///
+/// You should used this fixed splitter instead of the fixed splitter if the
```
Comment (natebosch) [inline third_party/packages/split/lib/split.dart]: Did you intend for the first "fixed splitter" to be "flex splitter"?

File: third_party/packages/split/lib/split.dart [hunk 4]
```diff
@@ -125,3 +193,64 @@ StreamSubscription<Object> flexSplitBidirectional(
     createSplitter();
   });
 }
+
+/// Creates a flex splitter that changes from horizontal to vertical depending
+/// on the window aspect ratio.
+///
+/// You should used this fixed splitter instead of the fixed splitter if the
```
Comment (natebosch) [inline third_party/packages/split/lib/split.dart]: First "fixed" to "flex"?

File: third_party/packages/split/lib/split.dart [hunk 5]
```diff
@@ -125,3 +193,64 @@ StreamSubscription<Object> flexSplitBidirectional(
...
+/// size of the parent element of the element being split isn't fixed. Keep in
+/// mind that the children being split must be sized such that flex-shrink does
+/// not apply as otherwise flex-shrink will interact badly with the calculation
+/// for the size of the two split regions.
+/// https://developer.mozilla.org/en-US/docs/Web/CSS/flex-shrink
+///
+/// [parts] must be a list of [CoreElement], [Element], or query selectors.
```
Comment (natebosch) [inline third_party/packages/split/lib/split.dart]: How would a query selector fit the `Element` list type?

File: third_party/packages/split/lib/split.dart (hunk 3 above)
Comment (kenzieschmoll) [inline third_party/packages/split/lib/split.dart]: nit: typo - I think it should read "You should use this flex spitter"

File: third_party/packages/split/lib/split.dart [hunk 6]
```diff
@@ -40,11 +42,22 @@ class _SplitOptions {
   external List<num> get sizes;
 
   external List<num> get minSize;
+
+  /// minSize is ignored unless this property is set.
+  external bool get expandToMin;
```
Comment (kenzieschmoll) [inline third_party/packages/split/lib/split.dart]: nit: optional - maybe rename to `splitAtMin`. Expanding to a minimum just seems a little funky. Or maybe `collapseToMin`

File: third_party/packages/split/lib/split.dart [hunk 7]
```diff
@@ -88,31 +107,80 @@ Splitter flexSplit(
...
       minSize: minSize,
+      expandToMin: minSize?.isNotEmpty == true,
     ),
   );
 }
 
-/// Creates a splitter that changes from horizontal to vertical depending on the
-/// window aspect ratio.
+/// Splitter that splits multiple elements that must have a parent of fixed
+/// width.
```
Comment (kenzieschmoll) [inline third_party/packages/split/lib/split.dart]: nit: replace width with size. The splitter could be splitting elements vertically or horizontally

File: third_party/packages/split/lib/split.dart [hunk 8]
```diff
@@ -88,31 +107,80 @@ Splitter flexSplit(
...
+/// schemes other than flexbox but we don't need that flexibility.
+Splitter fixedSplit(
   List<Element> parts, {
+  bool horizontal = true,
   gutterSize = 5,
-  List<num> verticalSizes,
-  List<num> horizontalSizes,
+  List<num> sizes,
   List<num> minSize,
+  expandToMin: true,
```
Comment (kenzieschmoll) [inline third_party/packages/split/lib/split.dart]: same naming comment as above

File: third_party/packages/split/lib/split.dart [hunk 9]
```diff
@@ -88,31 +107,80 @@ Splitter flexSplit(
...
+}) {
+  return _split(
+    parts,
+    _SplitOptions(
+      elementStyle: allowInterop((dimension, size, gutterSize, index) {
+        Object o = js_util.newObject();
+        js_util.setProperty(o, horizontal ? 'width' : 'height',
+            'calc($size% - ${gutterSize}px)');
```
Comment (kenzieschmoll) [inline third_party/packages/split/lib/split.dart]: trailing comma

File: third_party/packages/split/lib/split.dart (hunk 4 above)
Comment (kenzieschmoll) [inline third_party/packages/split/lib/split.dart]: nit: typo in first sentence

File: third_party/packages/split/lib/split.dart [hunk 10]
```diff
@@ -125,3 +193,64 @@ StreamSubscription<Object> flexSplitBidirectional(
...
+    parts,
+    gutterSize: gutterSize,
+    verticalSizes: verticalSizes,
+    horizontalSizes: horizontalSizes,
+    minSize: minSize,
+    splitterBuilder: flexSplit,
+  );
+}
+
+/// Creates a flex splitter that changes from horizontal to vertical depending
```
Comment (kenzieschmoll) [inline third_party/packages/split/lib/split.dart]: nit: r/flex/fixed

File: third_party/packages/split/lib/split.dart (hunk 6 above)
Comment (jacob314) [reply third_party/packages/split/lib/split.dart]: these are names from the JS library so I can't change them.

File: third_party/packages/split/lib/split.dart (hunk 1 above)
Comment (jacob314) [reply third_party/packages/split/lib/split.dart]: done.

File: third_party/packages/split/lib/split.dart (hunk 3 above)
Comment (jacob314) [reply third_party/packages/split/lib/split.dart]: done

Comment (jacob314) [reply third_party/packages/split/lib/split.dart]: done

File: third_party/packages/split/lib/split.dart (hunk 2 above)
Comment (jacob314) [reply third_party/packages/split/lib/split.dart]: thanks for the link. I'll adopt the standard style for this.

File: third_party/packages/split/lib/split.dart (hunk 7 above)
Comment (jacob314) [reply third_party/packages/split/lib/split.dart]: done

File: third_party/packages/split/lib/split.dart (hunk 8 above)
Comment (jacob314) [reply third_party/packages/split/lib/split.dart]: removed

File: third_party/packages/split/lib/split.dart (hunk 9 above)
Comment (jacob314) [reply third_party/packages/split/lib/split.dart]: done

File: third_party/packages/split/lib/split.dart (hunk 4 above)
Comment (jacob314) [reply third_party/packages/split/lib/split.dart]: done

Comment (jacob314) [reply third_party/packages/split/lib/split.dart]: done

File: third_party/packages/split/lib/split.dart (hunk 5 above)
Comment (jacob314) [reply third_party/packages/split/lib/split.dart]: Removed all the comments on [parts] as clearly at some point in the past the type of the List was changed but the comment wasn't removed. This is copy and paste from the comment from the other similar method in the file.

File: third_party/packages/split/lib/split.dart (hunk 10 above)
Comment (jacob314) [reply third_party/packages/split/lib/split.dart]: done

**Comment by kenzieschmoll** [comment]
lgtm_army

File: third_party/packages/split/lib/split.dart [hunk 11]
```diff
@@ -88,31 +107,86 @@ Splitter flexSplit(
...
+/// Splitter that splits multiple elements that must have a parent of fixed
+/// size.
+///
+/// You should used this fixed splitter instead of flex splitter when the parent
+/// of the elements being split has a fixed size but one or more of the children
+/// may have arbitrary size resulting in flex-shrink causing problems for the
+/// flex calculations.
+/// https://developer.mozilla.org/en-US/docs/Web/CSS/flex-shrink
 ///
 /// [parts] must be a list of [CoreElement], [Element], or query selectors.
```
Comment (natebosch) [inline third_party/packages/split/lib/split.dart]: Another copy of the stale comment

Comment (jacob314) [reply third_party/packages/split/lib/split.dart]: fixed all 3 other stale comments.

**Pull request merged**